### ISSUE TITLE
Importing sequence to sequence keras models into DL4J

### ISSUE THREAD

**Issue by drexjojo** [deeplearning4j/deeplearning4j#5549]
Hi,
I am trying to import a simple sequence-to-sequence (encoder-decoder) model into DL4J.
I get the following error : 
`INFO: Unexpected end-of-input: was expecting closing '"' for name
 at [Source: {"class_name": "Model", "config": {"name": "model_1", "layers": [{"name": "input_1", "class_name": "InputLayer", "config": {"name": "input_1", "dtype": "float32", "sparse": false, "batch_input_shape": [null, null]}, "inbound_nodes": []}, {"name": "embedding_1", "class_name": "Embedding", "config": {"embeddings_regularizer": null, "dtype": "float32", "mask_zero": false, "trainable": false, "name": "embedding_1", "embeddings_initializer": {"class_name": "RandomUniform", "config": {"minval": -0.05, "seed": null, "maxval": 0.05}}, "input_dim": 20000, "activity_regularizer": null, "batch_input_shape": [null, 20], "input_length": 20, "embeddings_constraint": null, "output_dim": 200}, "inbound_nodes": [[["input_1", 0, 0, {}]]]}, {"name": "bidirectional_1", "class_name": "Bidirectional", "config": {"name": "bidirectional_1", "merge_mode": "concat", "layer": {"class_name": "LSTM", "config": {"recurrent_constraint": null, "stateful": false, "trainable": true, "bias_initializer": {"class_name": "Zeros", "config": {}}, "activity_regularizer": null, "recurrent_activation": "hard_sigmoid", "recurrent_initializer": {"class_name": "Orthogonal", "config": {"seed": null, "gain": 1.0}}, "use_bias": true, "recurrent_regularizer": null, "kernel_constraint": null, "kernel_regularizer": null, "units": 300, "name": "lstm_1", "return_state": true, "recurrent_dropout": 0.0, "bias_constraint": null, "activation": "tanh", "unroll": false, "bias_regularizer": null, "return_sequences": false, "dropout": 0.0, "go_backwards": false, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"mode": "fan_avg", "seed": null, "scale": 1.0, "distribution": "uniform"}}, "implementation": 1, "unit_forget_bias": true}}, "trainable": true}, "inbound_nodes": [[["embedding_1", 0, 0, {}]]]}, {"name": "input_2", "class_name": "InputLayer", "config": {"name": "input_2", "dtype": "float32", "sparse": false, "batch_input_shape": [null, null, 4]}, "inbound_nodes": []}, {"name": "concatenate_1", "class_nam; line: 1, column: 4001]
Exception in thread "main" org.deeplearning4j.nn.modelimport.keras.exceptions.InvalidKerasConfigurationException: Keras LSTM layer accepts only one input (received 3). For more information, see http://deeplearning4j.org/model-import-keras.`

I can use the same JSON file to load the model into Keras and have checked that the JSON file is complete and does not have a trailing '"'

The full details of the model can be found here : [here](https://blog.keras.io/a-ten-minute-introduction-to-sequence-to-sequence-learning-in-keras.html)

**Comment by drexjojo** [comment]
The trained models can be found [here](https://drive.google.com/drive/folders/1S1tFHrglAcxLTbCYBJo7wGwtyCTHNv_m?usp=sharing)

**Comment by maxpumperla** [comment]
thanks for reporting, will look into this in a bit

**Comment by maxpumperla** [comment]
@drexjojo, could you please also give me your keras code? I want to better understand how this is defined in general.

![model](https://user-images.githubusercontent.com/3462566/41225225-a880a0a2-6d6e-11e8-9898-f560026c39bc.png)

In this situation, am I correct that you want `concatenate_1` to be the input to the LSTM and `concatenate_1` and `input_2` to be the `initial_state`s of the LSTM?

**Comment by drexjojo** [comment]
Hi @maxpumperla That is true, concatenate_1 and concatenate_2 are the initial_states of lstm_2 and input_2 is the input to lstm_2.
[Here](https://github.com/drexjojo/confusion_pair_correction/blob/master/train.py) is the code where I have defined the model.

**Comment by bjornamr** [comment]
What is the progress on this issue @maxpumperla ?

**Comment by maxpumperla** [comment]
I'll let you know once it's done. It's not very straightforward, see issue #5567 above. Didn't get around to that yet.
